frontend/src/modules/app/ToolBar.tsx

```diff
@@ -242,15 +243,25 @@ export const ToolBar = ({
   };
 
   /**
-   * Set Label for Start/Pause Ventilation button
+   * Enable/Disable and set label for Start/Pause Ventilation button
    */
   useEffect(() => {
-    if (ventilatingStatus) {
-      setLabel('Connecting...');
+    if (!ventilating) {
+      setIsDisabled(ventilatingStatusChanging || !firmwareConnected || !backendConnected);
+      if (!firmwareConnected || !backendConnected) {
+        setLabel(ventilatingStatusChanging ? 'Starting...' : 'Connecting...');
```

Comment:
This is a deviation from https://github.com/pez-globo/pufferfish-software/issues/361#issuecomment-867123868 , which specifies that when we're in standby and there's a disconnection, the button always shows "Connecting...". I've been going back and forth in my head about whether we want it to behave as specified in that table, or whether we want it to behave as you've implemented (which is how you specified it in row 4 of https://github.com/pez-globo/pufferfish-software/issues/361#issuecomment-866649898). I see the advantage of your approach, in the following sequence of events, an extreme edge case we hadn't discussed:

1. The firmware & backend are connected, and the ventilator is in standby mode.
2. The user presses the "Start Ventilation" button.
3. While the message is in transit, then the button changes to say "Starting..." and is disabled
4. While the message is still in transit, the firmware/backend becomes disconnected. If we follow your implementation, the button remains the same. If we follow the decision table I wrote, then while the button stays disabled the logic will automatically change the button to say "Connecting...". Either way, the user will see an alarm indicating loss of software connection.

Your implementation makes the behavior in standby mode more symmetric with the behavior in dashboard mode. Then we can express the logic as follows:

```
if (response.ventilating) {
  setIsDisabled(...)
  setLabel(ventilatingStatusChanging ? 'Pausing...' : 'Pause Ventilation') - note that your line 259 is redundant to line 262, except that line 262 is missing a third period in the "..."
  return;
}

setIsDisabled(...)
if (ventilatingStatusChanging) {
  setLabel('Starting...')
  return;
}

setLabel(firmwareConnected && backendConnected ? 'Start Ventilation' : 'Connecting...')
```

If you agree that we should follow your implementation, can you add an updated version of the decision table, and a link back to this comment, as a new comment in #361?

Reply:
> This logic is different from [#361 (comment)](https://github.com/pez-globo/pufferfish-software/issues/361#issuecomment-867123868) , which specifies that when we're in standby and there's a disconnection, the button should show "Connecting..." regardless of the value of request.ventilating. I've been going back and forth in my head about whether we want it to behave as specified in that table, or whether we want it to behave as you've implemented (which is how you specified it in row 4 of [#361 (comment)](https://github.com/pez-globo/pufferfish-software/issues/361#issuecomment-866649898)). I see the advantage of your approach, in the following sequence of events, an extreme edge case we hadn't discussed:

Honestly i had realised this last night and was going to fix it this morning haha!

> 1. The firmware & backend are connected, and the ventilator is in standby mode.
> 2. The user presses the "Start Ventilation" button.
> 3. While the message is in transit, then the button changes to say "Starting..." and is disabled
> 4. While the message is still in transit, the firmware/backend becomes disconnected. If we follow your implementation, the button remains the same. If we follow the decision table I wrote, then while the button stays disabled the logic will automatically change the button to say "Connecting...". Either way, the user will see an alarm indicating loss of software connection.
> Your implementation makes the behavior in standby mode more symmetric with the behavior in dashboard mode. 

But considering the case you mentioned and the code snippet that simplifies the logic, i agree that this is more symmetric to what we have in dashboard mode.


> If you agree that we should follow your implementation, can you add an updated version of the decision table, and a link back to this comment, as a new comment in #361?

Yes, Let's follow this? I'm adding the comment to the issue.